src/dask_awkward/layers/layers.py

```diff
@@ -188,9 +196,18 @@ def __repr__(self) -> str:
     @property
     def is_projectable(self) -> bool:
         # isinstance(self.io_func, ImplementsProjection)
-        return (
-            io_func_implements_projection(self.io_func) and not self.has_been_unpickled
-        )
+        if self._is_projectable is None:
+            return (
+                io_func_implements_projection(self.io_func)
+                and not self.has_been_unpickled
+            )
+        assert isinstance(self._is_projectable, bool)
```

Comment:
These ind of type-checks should really not be necessary. In fact, assertions should not normally appear in non-test code at all.

Reply:
yes :+1: I'll remove them